src/node_os.cc

```diff
@@ -290,6 +291,37 @@ static void GetHomeDirectory(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+static void GetUserInfo(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  uv_passwd_t pwd;
+
+  const int err = uv_os_get_passwd(&pwd);
+
+  if (err) {
+    return env->ThrowUVException(err, "uv_os_get_passwd");
+  }
+
+  Local<Object> entry = Object::New(env->isolate());
+
+  entry->Set(env->username_string(),
+             OneByteString(env->isolate(), pwd.username));
+
+  if (pwd.shell == NULL) {
+    entry->Set(env->shell_string(), Null(env->isolate()));
+  } else {
+    entry->Set(env->shell_string(),
+               OneByteString(env->isolate(), pwd.shell));
+  }
+
+  entry->Set(env->homedir_string(),
+             OneByteString(env->isolate(), pwd.homedir));
```

Comment:
Quick nit: given the likelihood that this is running on linux, there's a possibility that this may not actually be a OneByteString. It complicates things just a bit, but it might be worthwhile allowing an optional options (e.g. `{encoding:'utf8'}` to be passed in to `GetUserInfo` so that the encoding of the path can be specified. This would make it consistent with the new Buffer support in the fs module.